src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -428,105 +563,227 @@
                 {
-                    return false;
+                    if (attribute.IsConditionallyOmitted)
+                    {
+                        continue;
+                    }
+
+                    counts[attribute] = counts.TryGetValue(attribute, out var foundCount) ? foundCount + 1 : 1;
                 }
 
-                // The purpose of this check is to determine whether the interface parts of the member names agree,
-                // but to do so using robust symbolic checks, rather than syntactic ones.  Therefore, if neither member
-                // name contains an interface name, this check is not relevant.  
-                // Phrased differently, the explicitly implemented interface is not part of the signature unless it's
-                // part of the name.
-                if (sawInterfaceInName1)
+                foreach (var attribute in attributes2)
                 {
-                    Debug.Assert(sawInterfaceInName2);
+                    if (attribute.IsConditionallyOmitted)
+                    {
+                        continue;
+                    }
 
-                    // May avoid realizing interface members.
-                    if (member1.IsExplicitInterfaceImplementation() != member2.IsExplicitInterfaceImplementation())
+                    if (!counts.TryGetValue(attribute, out var foundCount) || foundCount == 0)
                     {
                         return false;
                     }
 
-                    // By comparing symbols, rather than syntax, we gain the flexibility of ignoring whitespace
-                    // and gracefully accepting multiple names for the same (or equivalent) types (e.g. "I<int>.M"
-                    // vs "I<System.Int32>.M"), but we lose the connection with the name.  For example, in metadata,
-                    // a method name "I.M" could have nothing to do with "I" but explicitly implement interface "I2".
-                    // We will behave as if the method was really named "I2.M".  Furthermore, in metadata, a method
-                    // can explicitly implement more than one interface method, in which case it doesn't really
-                    // make sense to pretend that all of them are part of the signature.
+                    counts[attribute] = foundCount - 1;
+                }
 
-                    var explicitInterfaceImplementations1 = member1.GetExplicitInterfaceImplementations();
-                    var explicitInterfaceImplementations2 = member2.GetExplicitInterfaceImplementations();
+                return counts.Values.All(c => c == 0);
+            }
 
-                    if (!explicitInterfaceImplementations1.SetEquals(explicitInterfaceImplementations2, SymbolEqualityComparer.ConsiderEverything))
-                    {
-                        return false;
-                    }
+            static ImmutableArray<ParameterSymbol> getParametersIncludingParameterOfExtension(Symbol member)
```

Comment:
>getParametersIncludingParameterOfExtension [](http://example.com/codeflow?start=51&length=42)

I am not convinced that we need to change this code to specially handle extension types. I think it just complicates things. Instead consider combining two comparisons:
- Compare the type in a regular way, this is going to take care of type parameters and everything else about them
- Compare the marker method, this is going to take care of the extension parameter and everything else on it.

Reply:
BTW, to compare type parameters on an extension we probably could simply call a static methos that takes just type parameters.